src/zip.js

```diff
@@ -22,28 +26,41 @@ const zipFunctions = async function (
     jsExternalModules = [],
     jsIgnoredModules = [],
     parallelLimit = DEFAULT_PARALLEL_LIMIT,
-    skipGo,
+    skipGo = true,
     zipGo,
   } = {},
 ) {
-  const [srcPaths, pluginsModulesPath] = await Promise.all([getSrcPaths(srcFolder), getPluginsModulesPath(srcFolder)])
+  await makeDir(destFolder)
 
+  const paths = await listFunctionsDirectory(srcFolder)
+  const functions = await runtimes.getFunctionsFromPaths(paths, { dedupe: true })
+  const pluginsModulesPath = await getPluginsModulesPath(srcFolder)
   const zipped = await pMap(
-    srcPaths,
-    (srcPath) =>
-      zipFunction(srcPath, destFolder, {
+    functions.values(),
+    async (func) => {
+      const zipResult = await func.runtime.zipFunction({
```

Comment:
Rather than calling the `zipFunction` function defined in this file, we're directly calling the `zipFunction` defined by each runtime. We can do this because during the discovery phase (i.e. lines 35-36) we've already collected all the information we need about the functions we're going to process.

The `zipFunction` function defined below remains as a top-level, entry point function for the application, so it includes the discovery step itself, in order to retrieve information about the function at the specific path. Doing this again from `zipFunctions` would be redundant.